tests/hash/generic_hash_helper.py

```diff
@@ -429,6 +432,7 @@ def get_interfaces_for_test(duthost, mg_facts, hash_field):
         for portchannel in portchannels.keys():
             if portchannel not in uplink_interfaces.keys():
                 downlink_interfaces.extend(portchannels[portchannel]['members'])
+    downlink_interfaces = [interface for interface in downlink_interfaces if interface not in uplink_interfaces]
```

Comment:
Why the `uplink_interfaces` are not removed from `downlink_interfaces` before this change?

Reply:
> Why the `uplink_interfaces` are not removed from `downlink_interfaces` before this change?

Because downlink_interfaces is generated by mg_facts['minigraph_interfaces'], it may contains upstream_interfaces.

>     downlink_interfaces = []
>     if mg_facts['minigraph_vlan_interfaces']:
>         vlan_interface = mg_facts['minigraph_vlan_interfaces'][0]['attachto']
>         downlink_interfaces = mg_facts['minigraph_vlans'][vlan_interface]['members']
>     elif mg_facts['minigraph_interfaces']:
>         for interface in mg_facts['minigraph_interfaces']:
>             downlink_interfaces.append(interface['attachto'])